File: SunnyFlamingo/ClientApp/src/app/models/good-model.ts
export class GoodModel{
  name: string;
  price: number;
  isAvailable: boolean;
  description: string;
  producerId: string;
  manufacturerId: string;
  material: string;
  color: string;
  imageIds: string[];
  constructor(options: {
    name: string;
    price: number;
    isAvailable: string;
    description: string;
    producerId: string;
    manufacturerId: string;
    material: string;
    color: string;
    imageIds: string[];
  }){
    this.name = options.name;
    this.price = +options.price;
    if (options.isAvailable === "true") {
      this.isAvailable = true;
    }
    else{
      this.isAvailable = false;
    }
    this.description = options.description;
    this.producerId = options.producerId;
    this.manufacturerId = options.manufacturerId;
    this.material = options.material;
    this.color = options.color;
    this.imageIds = options.imageIds;
  }
}

File: SunnyFlamingo/Models/GoodModel.cs
using SunnyFlamingo.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SunnyFlamingo.Models
{
    public class GoodModel
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public decimal Price { get; set; }
        public bool IsAvailable { get; set; }
        public string Description { get; set; }
        public DateTime AddTime { get; set; }

        public Guid? ProducerId { get; set; }
        public Guid? ManufacturerId { get; set; }
        public string Material { get; set; }
        public string Color { get; set; }
        public List<Guid> ImageIds { get; set; }
        public Guid? UserId { get; set; }
    }
}
